hooks/use-memoized-callback.ts

```
@@ -0,0 +1,17 @@
+import { useCallback, useRef } from 'react'
+
+export function useMemoizedCallback<T extends (...args: any[]) => any>(
+  callback: T,
+  deps: React.DependencyList
+): T {
+  const callbackRef = useRef<T>(callback)
+  const depsRef = useRef<React.DependencyList>(deps)
+
+  // Update callback if dependencies have changed
+  if (!depsRef.current || !deps.every((dep, index) => dep === depsRef.current![index])) {
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)